Fortnox.NET.Tests/Fortnox.NET.Tests/ContractTemplateTests.cs

```diff
@@ -0,0 +1,126 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using FortnoxNET.Communication;
+using FortnoxNET.Communication.ContractTemplates;
+using FortnoxNET.Models.ContractTemplates;
+using FortnoxNET.Services;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace FortnoxNET.Tests
+{
+    [TestClass]
+    public class ContractTemplateTests : TestBase
+    {
+        [TestMethod]
+        public async Task ItCanGetContractTemplates()
+        {
+            var request = new ContractTemplateListRequest(connectionSettings.AccessToken, connectionSettings.ClientSecret);
+            var response = await ContractTemplateService.GetContractTemplatesAsync(request);
+            
+            Assert.IsNotNull(response);
+            Assert.IsNotNull(response.Data);
+        }
+
+        [TestMethod]
+        public async Task ItCanGetAContractTemplate()
+        {
+            var templates = await GetTemplates();
+
+            if (!templates.Any())
+            {
+                Assert.Inconclusive("No contract templates exist in the system");
+                return;
+            }
+            
+            var request = new ContractTemplateListRequest(connectionSettings.AccessToken, connectionSettings.ClientSecret);
+            var response = await ContractTemplateService.GetContractTemplateAsync(
+                request,
+                templates.First().ContractTemplate
+            );
+            
+            Assert.IsNotNull(response);
+            Assert.IsNotNull(response.Data);
+            Assert.AreEqual(templates.First().ContractTemplate, response.Data.TemplateNumber);
+        }
+
+        // WARNING: Contract Templates have no DELETE endpoint, so running this can add extraneous data to the system
```

Comment:
Do we really want commented out tests? I think we should keep them available.

Reply:
It's important that the tests are re-runable, we don't want false negatives because of a broken data-set. Can we randomize things and keep the tests from being flaky?

Reply:
resolved